### PR TITLE
python312Packages.{ncclient, sshtunnel,docker}: fix missing dependency after paramiko changes

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

The two packages are broken since at least https://github.com/SuperSandro2000/nixpkgs/commit/a97420b26d3302913e5393e1cd888a069bd46089

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/paramiko/default.nix [hunk 1]
```diff
@@ -59,7 +62,7 @@ buildPythonPackage rec {
     mock
     pytestCheckHook
     pytest-relaxed
-  ] ++ lib.flatten (builtins.attrValues passthru.optional-dependencies);
```
Comment (dasJ) [inline pkgs/development/python-modules/paramiko/default.nix:62]: Why the first change?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/paramiko/default.nix:62]: yeah, I can send that on staging to reduce the rebuild count.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/paramiko/default.nix]: On the why:
- pyproject = true is the preferred way over format and all packages should slowly be converted
- optional-dependencies moved out of passthru a while ago
- the functions in lib are preferred over builtins but minor thingy

Comment (Conni2461) [reply pkgs/development/python-modules/paramiko/default.nix]: i think his question was only about changing `builtins.attrValues` to `lib.attrValues` as it is just an alias for `builtins`: https://github.com/NixOS/nixpkgs/blob/master/lib/attrsets.nix#L539

**Review by Conni2461 — APPROVED** [review]

File: pkgs/development/python-modules/docker/default.nix [hunk 2]
```diff
@@ -58,10 +57,7 @@ buildPythonPackage rec {
 
   nativeCheckInputs = [
     pytestCheckHook
-    paramiko
-    pynacl
-    websocket-client
-  ];
+  ] ++ lib.flatten (lib.attrValues optional-dependencies);
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/docker/default.nix:60]: see https://github.com/docker/docker-py/blob/7.1.0/.github/workflows/ci.yml#L50 and we want to follow upstream here

**Pull request merged**